src/extension.ts

```diff
@@ -674,12 +674,15 @@ export class ExtensionManager implements vscode.Disposable {
                 }
                 // Inform cpptools that custom CppConfigurationProvider will be able to service the current workspace.
                 this.ensureCppToolsProviderRegistered();
-                if (cpptools.notifyReady && this.cpptoolsNumFoldersReady < this.projectController.numOfWorkspaceFolders) {
-                    ++this.cpptoolsNumFoldersReady;
-                    if (this.cpptoolsNumFoldersReady === this.projectController.numOfWorkspaceFolders) {
-                        // Notify cpptools that the provider is ready to provide IntelliSense configurations.
-                        cpptools.notifyReady(this.configProvider);
-                        this.configProvider.markAsReady();
+                if (cpptools.notifyReady) {
+                    const projectCount = await this.projectController.getNumOfValidProjects();
+                    if (this.cpptoolsNumFoldersReady < projectCount) {
```

Comment:
I'm seriously debating removing this check though.  If one project is ready, maybe we should just turn the config provider on immediately.

Reply:
I also think we can just call `cpptools.notifyReady(this.configProvider)` if one project has a CMakeLists.txt `project.hasCMakeLists()`